src/app/api/cron/cron.ts

```diff
@@ -0,0 +1,92 @@
+import { Redis } from "@upstash/redis"
+import { CronJob } from "cron"
+import { addAnimeInRedis, logCronJob } from "./helper"
+import { fetchAnime } from "./anilist"
+import { createDiscussionThread } from "@projectforum/server/db/queries"
+import { discussionThreads } from "@projectforum/server/db/schema"
+import { db } from "@projectforum/server/db"
+import { title } from "process"
+import { eq } from "drizzle-orm"
+// import { NextResponse } from "next/server"
+
+export const fetchCache = "force-no-store"
+
+const anime: number[] = process.env.CRON_JOB_ANIME_ID_ARRAY?.split(",").map(Number) || []
+
+// Initialize Redis outside of the job function
+const redis = new Redis({
+  url: process.env.REDIS_URL,
+  token: process.env.REDIS_TOKEN
+})
+
+export const job = new CronJob("*/10 * * * *", async () => {
+  try {
+    // Check and add anime to Redis
+    anime.map(async (animeId) => {
+      const exists = await redis.exists(animeId.toString(10))
+      if (!exists) {
+        addAnimeInRedis(animeId)
+      }
+    })
+
+    const currentTime = Date.now()
+    // await Promise.all(
+    anime.map(async (animeId, idx) => {
+      const value = await redis
+        .get(animeId.toString(10))
+        .catch((err) => `${err instanceof Error ? err.stack : err}`)
+      if (value === null) {
+        logCronJob(`${animeId} returned NaN value ${value} index: ${idx}`)
+      }
+
+      const timestamp = Number(value)
+      const data = await fetchAnime(animeId)
+
+      if (timestamp - currentTime <= 10 * 60 * 100 && timestamp - currentTime >= 0) {
```

Comment:
Make sure to change this condition

Reply:
What happened when we used browser to test condition was that the currentTime was always the time when I initially opened my browser (and it never changed) and that's why we had trouble with the conditions in the end.